velox/functions/lib/ArrayShuffle.cpp

```diff
@@ -112,12 +117,41 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
-} // namespace
+std::vector<std::shared_ptr<exec::FunctionSignature>> arrayShuffleWithCustomSeedSignatures() {
+    return {exec::FunctionSignatureBuilder()
+              .typeVariable("T")
+              .returnType("array(T)")
+              .argumentType("array(T)")
+              .constantArgumentType("bigint")
+              .build()};
+}
+
+exec::VectorFunctionMetadata arrayShuffleMetadata() {
+  return exec::VectorFunctionMetadataBuilder()
+      .deterministic(false)
+      .build();
+}
 
-// Register function.
-VELOX_DECLARE_VECTOR_FUNCTION_WITH_METADATA(
-    udf_array_shuffle,
-    signatures(),
-    exec::VectorFunctionMetadataBuilder().deterministic(false).build(),
-    std::make_unique<ArrayShuffleFunction>());
+std::shared_ptr<exec::VectorFunction> makeArrayShuffle(
+    const std::string& name,
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const core::QueryConfig& config) {
+  return std::make_unique<ArrayShuffleFunction>(std::random_device{}());
+}
+
+std::shared_ptr<exec::VectorFunction> makeArrayShuffleWithCustomSeed(
```

Comment:
Does this function should move to `sparkql` directory? If that's the case, other similar function should obey this rule.

Reply:
hi @mbasmanova Which way do you think is better?

Reply:
This pattern seems fine to me.